luigi/server.py

```diff
@@ -73,6 +92,62 @@ def get_template_path(self):
         return pkg_resources.resource_filename(__name__, 'templates')
 
 
+class AllRunHandler(BaseTaskHistoryHandler):
+
+    def get(self):
+        # get task history db in the same path
+        engine = create_engine('sqlite:///.task_history.db', echo=True)
+        conn_sqlite = engine.connect()
+        query = "select id, name from tasks order by id desc"
+        task_list = conn_sqlite.execute(query)
+        tasknames = []
+        for task in task_list:
+            tasknames.append(task['name'])
+        self.render("menu.html", tasknames=tasknames)
+
+
+class SelectedRunHandler(BaseTaskHistoryHandler):
+
+    def get(self, name):
+        tasks = {}
+        chartResults = {}
+        tableResults = []
+        # get task history db in the same path
+        engine = create_engine('sqlite:///.task_history.db', echo=True)
+        conn_sqlite = engine.connect()
+        task_query = "select id, name from tasks"
+        history_query = "select id, task_id, event_name, ts from task_events"
```

Comment:
Maybe use the sqlalchemy ORM instead. Or simply use `find_all_by_name` and lazy load the events. http://docs.sqlalchemy.org/en/rel_0_9/orm/tutorial.html